app/jobs/scheduled_message_job.rb

```diff
@@ -12,11 +12,20 @@ def perform(message:, send_at:, callback_url:)
     # Locking to prevent race conditions between jobs
     message.reporting_relationship.update!(last_contacted_at: message.send_at)
 
-    SMSService.instance.send_message(
+    message_info = SMSService.instance.send_message(
       message: message,
```

Comment:
Why not send number_from, number_to, and body as separate parameters instead of the message object?